hathor/indexes/memory_deps_index.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2021 Hathor Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING, Dict, FrozenSet, Iterator, List, Set, Tuple
+
+from structlog import get_logger
+
+from hathor.indexes.deps_index import DepsIndex, get_validation_state
+from hathor.transaction import BaseTransaction
+from hathor.transaction.transaction_metadata import ValidationState
+from hathor.util import not_none
+
+if TYPE_CHECKING:  # pragma: no cover
+    from hathor.transaction.storage import TransactionStorage
+
+logger = get_logger()
+
+
+class _DirDepValue(Dict[bytes, ValidationState]):
+    """This class is used to add a handy method to values on dependency indexes."""
+
+    def is_ready(self) -> bool:
+        """True if all deps' validation are fully connected."""
+        return all(val.is_fully_connected() for val in self.values())
+
+
+class MemoryDepsIndex(DepsIndex):
+    def __init__(self):
+        self.log = logger.new()
+
+        # Direct and reverse dependency mapping (i.e. needs and needed by)
```

Comment:
There is a very brief general explanation (before the specific part) on `hathor.indexes.rocksdb_deps_indexes`. I'm thinking that can be the base for a general explanation on `hathor.indexes.deps_indexes`, this could either refer to the general explanation or have a copy of that brief explanation. What do you think?

Reply:
I've added an extended explanation including some examples.

Reply:
Great explanation, thanks!